pantab/_reader.py

```diff
@@ -83,3 +107,17 @@ def frames_from_hyper(
                     result[table] = _read_table(connection=connection, table=table)
 
     return result
+
+
+def frame_from_hyper_query(
+    database: Union[str, pathlib.Path],
+    query: str,
+    *,
+    hyper: Optional[tab_api.HyperProcess] = None,
+) -> pd.DataFrame:
+    """See api.rst for documentation."""
+    with tempfile.TemporaryDirectory() as tmp_dir, ensure_hyper_process(hyper) as hpe:
+        tmp_db = shutil.copy(database, tmp_dir)
```

Comment:
I can't remember why I added the `shutil.copy` on reads - seems like it could be a waste of disk space. Something to look into in the future